### DISCUSSION TITLE
Error when trying to use network_configuration for FARGATE tasks

### DISCUSSION THREAD

**Discussion by fsanzdev** [gruntwork-io/knowledge-base#151] [Help] [answered]
Hi, 

I'm having some issues when trying to use network_configuration for FARGATE tasks. This is the code I have:
```
  desired_number_of_tasks = "1"
  launch_type             = "FARGATE"
  network_mode            = "awsvpc"
  task_cpu                = 512
  task_memory             = 1024
  network_configuration   = {
    subnets                       = dependency.vpc.outputs.private_app_subnet_ids
    vpc_id                        = dependency.vpc.outputs.vpc_id
    security_group_rules          = {
      sg-service-name-port = {
        type                     = "ingress"
        from_port                = local.container_port
        to_port                  = local.container_port
        protocol                 = "tcp"
        # source_security_group_id = "sg-${local.service_name}"
        cidr_blocks              = [dependency.vpc.outputs.vpc_cidr_block]
      }
    }
    additional_security_group_ids = []
    assign_public_ip              = false
```

But it looks like both, `source_security_group_id` and `cidr_blocks` attributes are required by the module, when according to the SG rule documentation you can't have both, having an `Error: Conflicting configuration arguments` message when applying the plan.

Could you please confirm if I'm doing something wrong?

**Comment by gruntwork-discussions** [comment]
You must label this issue with at least one of the following labels before it can be answered:

* [r:terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)
* [r:terraform-aws-vpc](https://github.com/gruntwork-io/terraform-aws-vpc)
* [r:terraform-aws-architecture-catalog](https://github.com/gruntwork-io/terraform-aws-architecture-catalog)
* [r:terraform-aws-asg](https://github.com/gruntwork-io/terraform-aws-asg)
* [r:terraform-aws-beanstalk](https://github.com/gruntwork-io/terraform-aws-beanstalk)
* [r:terraform-aws-cache](https://github.com/gruntwork-io/terraform-aws-cache)
* [r:terraform-aws-ci](https://github.com/gruntwork-io/terraform-aws-ci)
* [r:terraform-aws-ci-pipeline-example](https://github.com/gruntwork-io/terraform-aws-ci-pipeline-example)
* [r:terraform-aws-cis-service-catalog](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog)
* [r:terraform-aws-data-storage](https://github.com/gruntwork-io/terraform-aws-data-storage)
* [r:terraform-aws-ecs](https://github.com/gruntwork-io/terraform-aws-ecs)
* [r:terraform-aws-eks](https://github.com/gruntwork-io/terraform-aws-eks)
* [r:terraform-aws-elk](https://github.com/gruntwork-io/terraform-aws-elk)
* [r:terraform-aws-kafka](https://github.com/gruntwork-io/terraform-aws-kafka)
* [r:terraform-aws-lambda](https://github.com/gruntwork-io/terraform-aws-lambda)
* [r:terraform-aws-load-balancer](https://github.com/gruntwork-io/terraform-aws-load-balancer)
* [r:terraform-aws-messaging](https://github.com/gruntwork-io/terraform-aws-messaging)
* [r:terraform-aws-mongodb](https://github.com/gruntwork-io/terraform-aws-mongodb)
* [r:terraform-aws-monitoring](https://github.com/gruntwork-io/terraform-aws-monitoring)
* [r:terraform-aws-openvpn](https://github.com/gruntwork-io/terraform-aws-openvpn)
* [r:terraform-aws-sam](https://github.com/gruntwork-io/terraform-aws-sam)
* [r:terraform-aws-security](https://github.com/gruntwork-io/terraform-aws-security)
* [r:terraform-aws-server](https://github.com/gruntwork-io/terraform-aws-server)
* [r:terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets)
* [r:terraform-aws-zookeeper](https://github.com/gruntwork-io/terraform-aws-zookeeper)
* [r:terragrunt](https://github.com/gruntwork-io/terragrunt)
* [r:terratest](https://github.com/gruntwork-io/terratest)
* [r:repo-copier](https://github.com/gruntwork-io/repo-copier)


[Community Guidelines](https://github.com/gruntwork-io/knowledge-base/discussions/1)
[Getting help](https://gruntwork.io/support/)
[Gruntwork documentation](https://www.gruntwork.io)

**Comment by denis256** [comment]
Hello,
if I matched correctly definition, in module is created `aws_security_group_rule` with `source_security_group_id` / `cidr_blocks` : 

https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/modules/services/ecs-service/main.tf#L126

Definition of `aws_security_group_rule` specify: 
  * cidr_blocks - (Optional) List of CIDR blocks. Cannot be specified with `source_security_group_id`

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule

**Comment by fsanzdev** [comment]
Thanks for the reply @denis256, but it looks like in GW code there is no conditional and according to the TF documentation you refer:

`source_security_group_id - (Optional) Security group id to allow access to/from, depending on the type. Cannot be specified with cidr_blocks, ipv6_cidr_blocks, or self.`

So it looks like both (cidr_blocks and source_security_group_id) can't be used at the same time, it looks like it has to be one or the other

**Comment by fsanzdev** [comment, marked as answer]
It was not quite clear looking at the code or the documentation, but all made sense after looking at this example https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/examples/for-learning-and-testing/services/ecs-service/main.tf

One of those fields has to be set to null (not just remove it)
